pkg/webhook/remotesecret_mutator_test.go

```diff
@@ -27,30 +27,47 @@ import (
 )
 
 func TestStoreUploadData(t *testing.T) {
-	rs := &api.RemoteSecret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "rs",
-			Namespace: "ns",
+	tests := map[string]*api.RemoteSecret{
+		"data": {
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "rs",
+				Namespace: "ns",
+			},
+			UploadData: map[string][]byte{
+				"a": []byte("b"),
+			},
 		},
-		UploadData: map[string]string{
-			"a": "b",
+		"stringData": {
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "rs2",
+				Namespace: "ns",
+			},
+			StringUploadData: map[string]string{
+				"a": "b",
+			},
 		},
 	}
 
-	storage := remotesecretstorage.NewJSONSerializingRemoteSecretStorage(&memorystorage.MemoryStorage{})
+	for name, rs := range tests {
```

Comment:
I know it is not line-saving :) but maybe I would prefer just defining the tests as they are instead of going through the map.

Reply:
This is a so-called "table-driven test". As far as I can tell, it's quite a common practice in Go.

Reply:
Well then thank you for broadening my horizons :)